src/resource.ts

```diff
@@ -0,0 +1,340 @@
+/* eslint-disable new-cap */
+import qs from 'qs';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { cloneDeep } from 'lodash';
+
+import Model from './model';
+import List from './models/List';
+
+/**
+ * @deprecated since 2.2.0
+ */
+export type ResourceCallback = (error: any, resource: any) => void;
```

Comment:
Thanks, I'm okay with deprecating the callback. Most people use `async / await` nowadays anyway.

Reply:
Oops, forgot to tell about the deprecation, but thanks for letting me know it is okay to add!